src/plot/series/abstract-series.js

```diff
@@ -59,12 +69,164 @@ class AbstractSeries extends PureRenderComponent {
 
   constructor(props) {
     super(props);
+    // Mouse Handlers
     this._seriesMouseOverHandler = this._seriesMouseOverHandler.bind(this);
     this._valueMouseOverHandler = this._valueMouseOverHandler.bind(this);
     this._seriesMouseOutHandler = this._seriesMouseOutHandler.bind(this);
     this._valueMouseOutHandler = this._valueMouseOutHandler.bind(this);
     this._seriesClickHandler = this._seriesClickHandler.bind(this);
     this._valueClickHandler = this._valueClickHandler.bind(this);
+    // Keyboard Handlers
+    this._seriesKeyUpHandler = this._seriesKeyUpHandler.bind(this);
+    this._valueKeyUpHandler = this._valueKeyUpHandler.bind(this);
+    this._seriesKeyPressHandler = this._seriesKeyPressHandler.bind(this);
+    this._valueKeyPressHandler = this._valueKeyPressHandler.bind(this);
+    this._seriesKeyDownHandler = this._seriesKeyDownHandler.bind(this);
+    this._valueKeyDownHandler = this._valueKeyDownHandler.bind(this);
+    this._seriesFocusHandler = this._seriesFocusHandler.bind(this);
+    this._valueFocusHandler = this._valueFocusHandler.bind(this);
+    this._seriesBlurHandler = this._seriesBlurHandler.bind(this);
+    this._valueBlurHandler = this._valueBlurHandler.bind(this);
+  }
+
+  /**
+   * Focus handler for the specific series' value.
+   * @param {Object} d Value object
+   * @param {Object} event Event.
+   * @protected
+   */
+  _valueFocusHandler(d, event) {
```

Comment:
there is a lot of repeated code here. can this be dried up into a more general function creator so we dont have quite so many function polluting the space?